### PR TITLE
Translate text input

### PR DESCRIPTION
**Pull request by brianseek**
## Chromatic

https://{{head.ref}}--60f9b557105290003b387cd5.chromatic.com

## Description
Closes [va.gov-team/39885](https://app.zenhub.com/workspaces/check-in-experience-61fc23a2cb8a14001132e102/issues/department-of-veterans-affairs/va.gov-team/39885)

Is there a DS ticket related to this?

## Testing done
Updated existing text input tests.

## Screenshots
chrome-capture-2022-3-28

Screen Shot 2022-04-28 at 4 27 38 PM
Screen Shot 2022-04-28 at 5 11 14 PM

## Acceptance criteria
- [ ] Hard coded text for Error is translatable
- [ ] Hard coded text for Required is translatable
- [ ] Hard coded text for Max Characters is translatable

## Definition of done
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)

### CODE REVIEW

**Comment by pdavies88** [comment]
An additional prop for `va-text-input` will be added in #403 it is for `minLength`. There are no translations at this time for this prop and the message it outputs therefore it is not recommended to be utilized by this component until translations can be provided.

File: packages/storybook/stories/TextInput.stories.jsx [hunk 1]
```diff
@@ -1,6 +1,6 @@
-import React, { useState } from 'react';
+import React, { useEffect, useState } from 'react';
 
-import TextInput from '../../react-components/src/components/TextInput/TextInput';
+import { TextInput } from '@department-of-veterans-affairs/component-library';
```
Comment (bkjohnson) [inline packages/storybook/stories/TextInput.stories.jsx:3]: I haven't been able to pull your branch and test this, but I'm betting that if you go to the docs page for this story you'll see the prop table is empty. We probably don't want these changes committed.

File: packages/web-components/src/components/va-text-input/va-text-input.tsx [hunk 2]
```diff
@@ -148,6 +156,12 @@ export class VaTextInput {
     }
   };
 
+  connectedCallback() {
+    i18next.on('languageChanged', () => {
+      forceUpdate(this.el);
+    });
+  };
```
Comment (bkjohnson) [inline packages/web-components/src/components/va-text-input/va-text-input.tsx:163]: Lets have a `i18next.off('languageChanged')` in a `disconnectedCallback` as well.

**Review by bkjohnson — COMMENTED** [review]
**Edit**: I just realized that this PR is from a fork on your account rather than a branch in this repo, my bad. I'm guessing that's also why some of our CI isn't working. Do you not have permissions to write to a branch here?

Also, on your fork can you update this:

https://github.com/department-of-veterans-affairs/component-library/blob/8bde699994a1924420b6b6a779b23af03e80c914/.github/workflows/chromatic.yml#L5

to include this:

https://github.com/department-of-veterans-affairs/component-library/blob/8bde699994a1924420b6b6a779b23af03e80c914/.github/workflows/node.js.yml#L8-L9

That way we may be able to get a storybook deployment without cloning your fork.

---

In the future please don't use a colon in your branch name. It looks like it's broken our chromatic deployment, and it's also [not considered "acceptable" in git](https://git-scm.com/docs/git-check-ref-format#_description):

> They cannot have ASCII control characters (i.e. bytes whose values are lower than \040, or \177 DEL), space, tilde ~, caret ^, **or colon `:` anywhere**.

git also doesn't find this branch on the remote when I try to pull

File: packages/web-components/src/components/va-text-input/va-text-input.tsx (hunk 2 above)
Comment (brianseek) [reply packages/web-components/src/components/va-text-input/va-text-input.tsx:163]: added

**Comment by brianseek** [comment]
> **Edit**: I just realized that this PR is from a fork on your account rather than a branch in this repo, my bad. I'm guessing that's also why some of our CI isn't working. Do you not have permissions to write to a branch here?
> 
> Also, on your fork can you update this:
> 
> https://github.com/department-of-veterans-affairs/component-library/blob/8bde699994a1924420b6b6a779b23af03e80c914/.github/workflows/chromatic.yml#L5
> 
> to include this:
> 
> https://github.com/department-of-veterans-affairs/component-library/blob/8bde699994a1924420b6b6a779b23af03e80c914/.github/workflows/node.js.yml#L8-L9
> 
> That way we may be able to get a storybook deployment without cloning your fork.
> 
> **Edit 2**: It looks like we [can't deploy to chromatic from forked PRs](https://www.chromatic.com/docs/github-actions#forked-repositories) unless we want to make our project token public. @k80bowman should we make that public, or start a new PR on a branch from the repo?
> 
> ~In the future please don't use a colon in your branch name. It looks like it's broken our chromatic deployment, and it's also [not considered "acceptable" in git](https://git-scm.com/docs/git-check-ref-format#_description):~
> 
> ~> They cannot have ASCII control characters (i.e. bytes whose values are lower than \040, or \177 DEL), space, tilde `~`, caret ^, **or colon `:` anywhere**.~
> 
> ~git also doesn't find this branch on the remote when I try to pull~

@bkjohnson yep, I don't have access to create a PR directly so I did it from a forked repo. Perhaps its easier for you or someone on your team cherry-pick these commits into a new branch?? Let me know the best path, and anything I can do to help.

**Comment by bkjohnson** [comment]
I know this PR is older, but it looks like [4 days ago you were able to push directly to a branch](https://github.com/department-of-veterans-affairs/component-library/pull/411) so maybe you recently got added to the right team that has `write` permissions. If you want to do the cherry picking it looks like it's [possible to do it with a fork](https://gist.github.com/bhumphrey/3764983) - I'll close this so you can open a new PR.

**Pull request closed without merging**